render/vertexarray.h

```diff
@@ -11,11 +11,11 @@ class VertexArray
 {
 public:
     VertexArray();
-    ~VertexArray();
     VertexArray(const VertexArray&) = delete;
-    VertexArray(VertexArray&&) = delete;
     VertexArray& operator=(const VertexArray&) = delete;
-    VertexArray& operator=(VertexArray&&) = delete;
+    VertexArray(VertexArray&&);
+    VertexArray& operator=(VertexArray&&);
```

Comment:
So this is kind of interesting. Initally I had a lot of the copy/move constructors deleted for these OpenGL render classes. But I wanted to change the Texture for a Mesh to a stack variable instead, so I had to implement them all. However that caused a problem specfically when copying these objects.

Because they have a mRendererId that is created when the object on the GPU side is created. 
`glGenBuffers(1, &mRendererID);`
The destructor frees that object on the GPU side as well using that id
`glDeleteBuffers(1, &mRendererID);`

If the object is copied, the object it was copied from may destruct, and free the object from under us. So the only way for it to work without serious problems is to only implement the move constructors, and leave the copy constructors deleted. Because the move constructor will try to delete this object with an id of 0 which won't do anything
```
mRendererID = other.mRendererID;
other.mRendererID = 0;
```

Reply:
Good catch, and reasonable solution!

I guess the way to make the compy constructor would be to create a new object with a new ID for the new object, and not ever share the old ID